editors/sc-ide/core/settings/serialization.cpp

```diff
@@ -51,6 +51,19 @@ struct IODeviceSource : boost::iostreams::source {
     QIODevice *mDev;
 };
 
+static QColor yamlNodeToColor(const YAML::Node & node) {
+    // Official way of making an invalid color -- see http://doc.qt.io/qt-5/qcolor.html#QColor
+    QColor invalid = QColor();
+    if (!node) {
+        return invalid;
+    }
+    if (!node.IsScalar()) {
+        return invalid;
+    }
+    auto color_string = node.as<std::string>().c_str();
+    return QColor(color_string);
```

Comment:
this is a dangling pointer. at the end of evaluating the expression `node.as<std::string>().c_str()`, the `std::string` will be destructed, but the `const char*` referred to by `color_string` will be pointing into it. you can get around this by either extending the lifetime of the std::string:

```cpp
auto color_string = node.as<std::string>();
return QColor(color_string.c_str());
```

or eliminating the temporary (the std::string will not be destructed until the entire expression has been evaluated)

```cpp
return QColor(node.as<std::string>().c_str());
```

since i'm reviewing this anyway:
- generally better to put the 'success path' first
- no need to be so verbose with the conditions

```cpp
if (node && node.IsScalar()) {
    return QColor(node.as<std::string>().c_str());
} else {
    return QColor(); // invalid if default-constructed
}
```